doc/recipe/preprocessor.rst

```diff
@@ -901,24 +941,89 @@ days in a year may vary between calendars, (sub-)daily data are not supported.
 .. code-block:: yaml
 
     preprocessors:
-      multi_model_preprocessor:
+      multi_model_save_input:
         multi_model_statistics:
           span: overlap
           statistics: [mean, median]
           exclude: [NCEP]
+            multi_model_save_input:
+      multi_model_without_saving_input:
+        multi_model_statistics:
+          span: overlap
+          statistics: [mean, median]
+          exclude: [NCEP]
+          keep_input_datasets: false
 
-see also :func:`esmvalcore.preprocessor.multi_model_statistics`.
+Multi-model statistics now also supports a ``groupby`` argument. You can group by
+any dataset key (``project``, ``experiment``, etc.) or a combination of keys in a list. You can
+also add an arbitrary tag to a dataset definition and then group by that tag. When
+using this preprocessor in conjunction with `ensemble statistics`_ preprocessor, you
+can group by ``ensemble_statistics`` as well. For example:
 
-When calling the module inside diagnostic scripts, the input must be given
-as a list of cubes. The output will be saved in a dictionary where each
-entry contains the resulting cube with the requested statistic operations.
+.. code-block:: yaml
 
-.. code-block::
+    datasets:
+      - {dataset: CanESM2, exp: historical, ensemble: "r(1:2)i1p1"}
+      - {dataset: CCSM4, exp: historical, ensemble: "r(1:2)i1p1"}
 
-    from esmvalcore.preprocessor import multi_model_statistics
-    statistics = multi_model_statistics([cube1,...,cubeN], 'overlap', ['mean', 'median'])
-    mean_cube = statistics['mean']
-    median_cube = statistics['median']
+    preprocessors:
+      example_preprocessor:
+        ensemble_statistics:
+          statistics: [median, mean]
+        multi_model_statistics:
+          span: overlap
+          statistics: [min, max]
+          groupby: [ensemble_statistics]
+          exclude: [NCEP]
+
+This will first compute ensemble mean and median, and then compute the multi-model
+min and max separately for the ensemble means and medians. Note that this combination
+will not save the individual ensemble members, only the ensemble and multimodel statistics results.
```

Comment:
Move this paragraph to `ensemble_statistics`

Reply:
When `ensemble_statistics` is described, `multi_model_statistics` has not been described yet so I thought it would fit better in here. But whatever it can be moved